aten/src/ATen/ops/from_blob.h

```diff
@@ -85,6 +91,11 @@ class TORCH_API TensorMaker {
   std::unique_ptr<void, ContextDeleter> ctx_{nullptr, detail::noopDelete};
   c10::optional<Device> device_{};
   TensorOptions opts_{};
+  // Allows the storage of this tensor to be resized
+  // Note: this is won't actually resize anything (we use a meta allocator).
+  // This is only useful because at::for_blob is used by wrapper tensor subclasses,
+  // which we need to be able to advertise as resizeable.
```

Comment:
I'm not sure about this comment? It seems to be written with the specific wrapper subclass use case of this API in mind.
"Allows the storage of the created tensor to be resized" sounds like an accurate description to me.
Since resizeable is a known concept for Tensor/Storage, I'm not sure that warrants a comment here.

Reply:
So, the problem with resizing an `from_blob` tensor is we don't know how the original tensor was allocated, and so we cannot reallocate it in an equivalent way. This can matter if, e.g., the tensor is on shared/pinned memory.  For meta storage, however, it doesn't really matter, because there is no nontrivial allocator involved.

I guess you can kind of use this API if you don't mind us using the default allocator to service your resize. This might be the most reasonable semantics for this outside of Meta too.

Reply:
I originally planned on making this more limited - `resizeable_storage()` will **never** let you use a real allocator, and is just useful for flipping metadata (important for subclasses).

Is the argument that we want to something like `resizeable_storage(/*meta=*/true/false)`, so real users of `for_blob()` can indicate that they want us to attempt to use our default allocator? I hesitated to do that since there isn't a concrete use case that I know of, while the meta-allocator version has a more concrete use case. But open to either

Reply:
In that case, should we request an allocator to be able to set resizable? And it is the user responsability to privide both.
We can then make _make_wrapper_subclass pass the Meta allocator as we know it is what we want there.

Reply:
sure - so concretely, instead of a `.resizeable_storage()` API, we have a `.allocator(my_alloc)` API that lets you pass in a custom allocator, where that allocator's resizeable-ness is what determines whether or not our storage will be resizeable? And I'll pass the meta allocator in from `_make_wrapper_subclass`.

Reply:
You will need 2 APIs actually: one to pass in an allocator and one to say if this particular Tensor is resizable or not.

Reply:
There's not much point in passing an allocator if it's not resizable though 📦